### ISSUE TITLE
MemoryError

### ISSUE THREAD

**Issue by binggan-1-ui** [kalpitthakkar/pb-gcn#7]
Hello, there is a MemoryError in the runtime. What is the cause of this situation? I have 32 GB of memory and run time memory utilization is almost 100%.

**Comment by kalpitthakkar-lm** [comment]
@binggan-1-ui Can you give me some more information?

What is the memory available per GPU that you are using?
What's your batch size?

I tried running my network with a batch size of 4 on 4 Titan Xp. It occupies ~1G of memory per GPU when the training starts and this memory profile persists. A batch of 64 on 4 Titan Xp shows a memory occupancy of ~9G per GPU (that is, ~36G in total).

So, if you are using a batch size of 64, you probably would need more than 32G of memory (in total) at full floating-point precision.

**Comment by kalpitthakkar** [comment]
@binggan-1-ui Can you give me some more information?

What's the memory per GPU available?
What is your batch size?

I tried running my network with a batch size of 4 on 4 Titan Xp. It occupies ~1G of memory per GPU when the training starts and this memory profile persists. A batch of 64 on 4 Titan Xp shows a memory occupancy of ~9G per GPU (that is, ~36G in total).

So if you are using a batch size of 64, please make sure you have at least 36G available (assuming you use full floating point precision).

**Comment by lxy741** [comment]
Hello, there is a problem that I cannot fix it . as follows:
[ Tue Sep 29 16:21:16 2020 ] Training epoch: 1
D:\ProgramData\Anaconda3\lib\site-packages\numpy\core\fromnumeric.py:3118: RuntimeWarning: Mean of empty slice.
  out=out, **kwargs)
D:\ProgramData\Anaconda3\lib\site-packages\numpy\core\_methods.py:85: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)
[ Tue Sep 29 16:21:16 2020 ]    Mean training loss: nan.
[ Tue Sep 29 16:21:16 2020 ]    Time consumption: [Data]33%, [Network]33%
[ Tue Sep 29 16:21:16 2020 ] Training epoch: 2
[ Tue Sep 29 16:21:16 2020 ]    Mean training loss: nan.
[ Tue Sep 29 16:21:16 2020 ]    Time consumption: [Data]33%, [Network]33%
[ Tue Sep 29 16:21:16 2020 ] Training epoch: 3
[ Tue Sep 29 16:21:16 2020 ]    Mean training loss: nan.
[ Tue Sep 29 16:21:16 2020 ]    Time consumption: [Data]33%, [Network]33%
[ Tue Sep 29 16:21:16 2020 ] Training epoch: 4
[ Tue Sep 29 16:21:16 2020 ]    Mean training loss: nan.
[ Tue Sep 29 16:21:16 2020 ]    Time consumption: [Data]33%, [Network]33%
[ Tue Sep 29 16:21:16 2020 ] Training epoch: 5
[ Tue Sep 29 16:21:16 2020 ]    Mean training loss: nan.
[ Tue Sep 29 16:21:16 2020 ]    Time consumption: [Data]33%, [Network]33%
[ Tue Sep 29 16:21:16 2020 ] Eval epoch: 5
#####################################

**Comment by kalpitthakkar** [comment]
@lxy741 I need more information. What do you mean?
Can you show me your changes to the code that I have provided?
